### PR TITLE
BUG: Series.drop() with MultiIndex: inconsistent behaviour

### LINKED ISSUE #8594: Series.drop() with MultiIndex: inconsistent behaviour
The behaviour below occurs in versions `'0.15.0rc1-21-g32c5016'` and `'0.14.1'`.

When the `label` passed to the `drop` method of a `Series` is not in the axis:
(a) if the index is not a `MultiIndex` then an exception is raised (in most cases; see issue #8530)
(b) if the index is a `MultiIndex` then an exception is not raised; the original series is returned

Examples of current behaviour:
(a)

``` python
>>> pd.Series([10, 20, 30], [1, 2, 3]).drop(7)

ValueError: labels [7] not contained in axis
```

(b)

``` python
>>> pd.Series(
    [10, 20, 30],
    pd.MultiIndex(
        levels=[[1, 2, 3], [4, 5, 6]],
        labels=[[0, 1, 2], [0, 1, 2]],
        names=['a', 'b'])
).drop(7, level='a')

a  b
1  4    10
2  5    20
3  6    30
dtype: int64
```

I propose that in (b) an exception be raised. Furthermore, an exception should be raised in (b) if (for instance) the label passed to `drop` had been `4`, because although `4` is in the axis, it is not in the specified level.

The exception message in (b) would have to refer to the level, e.g.:

``` python
ValueError: labels [7] not contained in level
```

### PR DESCRIPTION
**Pull request by proost**
- [x] closes #8594
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

description:
In core.base 
```
        indexer = self.get_indexer(labels)
        mask = indexer == -1
        if mask.any():
            if errors != "ignore":
                raise KeyError(f"{labels[mask]} not found in axis")
```
single index check and raise if there is no matching indexer. But 'MultiIndex' case,
In indexes.multi
```
        codes = com.index_labels_to_array(codes)
        i = self._get_level_number(level)
        index = self.levels[i]
        values = index.get_indexer(codes)

        mask = ~algos.isin(self.codes[i], values)

        return self[mask]
```
no check. So i add check logic and raise exception if there is no matching indexer.
This bug cause from 'MultiIndex', so do not distinguish between "Series" and "DataFrame"

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -784,7 +784,7 @@ MultiIndex
 ^^^^^^^^^^
 
 - Constructior for :class:`MultiIndex` verifies that the given ``sortorder`` is compatible with the actual ``lexsort_depth``  if ``verify_integrity`` parameter is ``True`` (the default) (:issue:`28735`)
--
+- Series and MultiIndex `.drop` with `MultiIndex` raise exception if labels not in given level (:issue:`8594`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v1.0.0.rst]:
```suggestion
- Series and MultiIndex `.drop` with `MultiIndex` raise exception if labels not in given in level (:issue:`8594`)
```

File: pandas/tests/indexes/multi/test_drop.py [hunk 2]
```diff
@@ -139,3 +139,22 @@ def test_drop_not_lexsorted():
...
+    with pytest.raises(KeyError, match=msg):
+        s.drop(4, level="a")
+    with pytest.raises(KeyError, match=msg):
+        df.drop(4, level="a")
+
+    msg = r"labels \[7\] not found in level"
+    with pytest.raises(KeyError, match=msg):
+        s.drop(7, level="b")
+    with pytest.raises(KeyError, match=msg):
+        df.drop(7, level="b")
```
Comment (gfyoung) [inline pandas/tests/indexes/multi/test_drop.py]: We can probably parameterize this (via `pytest` idiom).  Not necessary to do now though.

File: pandas/tests/indexes/multi/test_drop.py [hunk 3]
```diff
@@ -139,3 +139,22 @@ def test_drop_not_lexsorted():
...
         tm.assert_index_equal(lexsorted_mi.drop("a"), not_lexsorted_mi.drop("a"))
+
+
+@pytest.mark.parametrize(
+    "msg,labels,level",
+    [
+        (r"labels \[4\] not found in level", 4, "a"),
+        (r"labels \[7\] not found in level", 7, "b"),
+    ],
+)
```
Comment (jreback) [inline pandas/tests/indexes/multi/test_drop.py]: can you add some test cases for ``errors='ignore'``

**Review by jreback — CHANGES_REQUESTED** [review]
looks good. ping on green.

**Comment by jreback** [comment]
also needs a rebase

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @proost

**Pull request merged**